core/ecschema-editing/src/Validation/SchemaComparer.ts

```diff
@@ -838,4 +859,37 @@ export class SchemaComparer {
 
     await Promise.all(promises);
   }
+
+  /**
+   * Compares properties with different full name that could potentially be the same in the context of comparing different schemas.
+   * @returns flag to indicate whether to report diagnostic or not.
+   */
+  private reportDiagnosticFlag(fullNameA: string, fullNameB: string, itemA: SchemaItem | AnyProperty | undefined, itemB: SchemaItem | AnyProperty | undefined): boolean {
```

Comment:
Put yourself in the shoes of someone reading the code who does not know it and tell me what you think this line does:
`const reportFlag = this.reportDiagnosticFlag(structNameA ?? "", structNameB ?? "", propertyA, propertyB);`

Could the code be written in a way that gave the user a better idea of it's function?


Now think about someone who has to write this code, could it be more succinct easier to call?


I believe you could refactor this method to be clearer to read, easier to call and avoid parsing strings.

Reply:
The function has been removed and replaced with `areItemsSameByName`.